WA3CRUD/WA3CRUD/Controllers/MarketController.cs

```diff
@@ -0,0 +1,122 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using WA3CRUD.Models;
+
+namespace WA3CRUD.Controllers
+{
+    [Route("api")]
+    [ApiController]
+    public class MarketController : ControllerBase
+    {
+        static readonly IMarketable phonemarket = new Market();
+
+
+        [HttpGet("Products")]
+        public IEnumerable<Product> GetAllProducts()
+        {
+            return phonemarket.GetAll();
+        }
+
+
+        [HttpGet("Products/id")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(Product))]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public IActionResult GetProduct(int id)
+        {
+            Product item = phonemarket.Get(id);
+            if (item == null)
+            {
+                return NotFound();
+            }
+            return Ok(item);
+        }
+
+
+        /*          
+        NOT WORKING variant
+                  
+        [HttpGet("Products/id")]
+        public Product GetProduct(int id)
+        {
+            Product item = phonemarket.Get(id);
+            if (item == null)
+            {
+                GetProductError();
+            }
+            return item;
+        }
+
+        [Route("Products/id")]
+        [HttpPost]
+        public IActionResult GetProductError()
+        {
+            return NotFound();
+        }
+        */
+
+        [HttpGet("Products/Manufacturer")]
+        public IEnumerable<Product> GetProductsByManufacturer(string company)
```

Comment:
if you use 'company' to navigate to the resource then you should get it from route